### PR TITLE
Table export csv [REOPEN]

### PR DESCRIPTION
**Pull request by daniel-sudz**
@Zjjc123 use an escaping library for the csv like the one in the firebase folder or it will break on special characters

### CODE REVIEW

File: website/src/Components/Table/exporter/handleExport.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const handleExport = (data: initialStateType) => {
+    let exportString = "Name,Description,Hours,Date\n";
+    Object.keys(data.data).forEach((entry) => {
+        const dateObj = JSON.parse(data.data[entry].Date) as Day;
+        exportString += `"${data.data[entry].Name}",`;
+        exportString += `"${data.data[entry].Description}",`;
+        exportString += `${data.data[entry].Hours},`;
+        exportString += `${dateObj.month}-${dateObj.day}-${dateObj.year}`  + ', ';
+        exportString += "\n";
```
Comment (daniel-sudz) [inline website/src/Components/Table/exporter/handleExport.ts:13]: this is not a good idea

Comment (daniel-sudz) [reply website/src/Components/Table/exporter/handleExport.ts:13]: This is what I used on the firebase side: 

```js
import stringify = require("csv-stringify/lib/sync");  // (use import syntax for react and you will need to download the modules)
const demoArray = [["header1", "header2"], ["content1", "content2"]] // etc..
const csv = stringify(demoArray); 
```

**Comment by daniel-sudz** [comment]
One thing that bugs me is that since we use listeners, we can't tell when it finished downloading data (at least I couldn't find an API for this). Basically this means that we have a potential race condition if the user pressed the download button before the data is available.  

You can mitigate by making the button unclickable when no data is present (which should reduce the race condition window significantly) but I don't think we can solve the problem fully. 

@Zjjc123

**Comment by Zjjc123** [comment]
@daniel-sudz I don't think the button is available until after the data loads

**Comment by Zjjc123** [comment]
@daniel-sudz I can't request review since you opened this one. Lmk when it's ready to merge

**Review by daniel-sudz — COMMENTED** [review]
Looks good @Zjjc123

**Pull request merged**